library/constant_time.c

```diff
@@ -46,6 +46,11 @@
 #endif
 
 #include <string.h>
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+#define PSA_TO_MBEDTLS_ERR( status ) PSA_TO_MBEDTLS_ERR_LIST( status,    \
```

Comment:
I don't know how handy it is to let developers to redefine a macro for every module. I mean, albeit this works, wouldn't it be better to have all these macros (this one and the others defined in various modules) all listed in `psa_util.h`?

Reply:
`MBEDTLS_ERR_` codes are different in every module. So `PSA_TO_MBEDTLS_ERR` has a different definition in every module. E.g. `PSA_ERROR_INSUFFICIENT_MEMORY` might map to `MBEDTLS_ERR_PK_ALLOC_FAILED`, `MBEDTLS_ERR_CIPHER_ALLOC_FAILED`, …

Having the same macro name everywhere allows moving code across modules, which I guess is a good thing.

Reply:
I see and I also agree with you. My point here was just that it doesn't look very handy to have to re-declare that define on top of each module and it's also error prone in case someone picks the wrong translation list.

What about what I proposed [here](https://github.com/Mbed-TLS/mbedtls/pull/6829#discussion_r1071066172)? 
I mean have a single error function like `psa_status_to_mbedtls`, but which takes an `enum` for the module that calls it instead of a vector. In this way the define on top of other modules would be something like
`#define PSA_TO_MBEDTLS_ERR_NAME   PSA_TO_MBEDTLS_ERR_xxx`
with `xxx` being the name of the module. Then the `psa_status_to_mbedtls` can pick the correct lookup table on its own. 
The advantage IMO is that for the define on top of the module you explicitly write the name of the module itself, so it would be easier to cross-check.

However this is only a design suggestion, not a real issue, so if it doesn't make much sense to you we can easily skip it ;)

Reply:
I thought about it, but I didn't want to add the small overhead of runtime checking for module, if we know it compile-time. In your proposal, instead of having one define at the top, each function call would need to pass a second parameter, so there's more overhead in code. Or, you could have a local macro that would cover that, and we're getting back to what we have here - per-module macro :)  
If what you mean is having local `#define PSA_TO_MBEDTLS_ERR PSA_TO_MBEDTLS_ERR_MODULE_NAME` and have the `PSA_TO_MBEDTLS_ERR_MODULE_NAME`'s all defined in `psa_util.h` - sure, that would work, but that's just a next layer of indirection that we have to dig in deeper to see what errors are used, and we still have a local define.
Side-note: there are also modules that use two different error sets (look [here](https://github.com/Mbed-TLS/mbedtls/pull/6829/files#diff-d7d2de928b7fbc94f41d2f947ae95745d7a89d85c27e08cd15cc37ef7c5ded37R44));

Reply:
Ok, I understand. So it's all OK for me ;)